pkg/util/containers/metrics/provider/registry_test.go

```diff
@@ -18,13 +20,13 @@ func TestCollectorRegistry(t *testing.T) {
 	assert.Nil(t, c.effectiveCollectors[RuntimeMetadata{runtime: RuntimeNameDocker}])
 
 	// Check for collectors (none are registered, should not change output)
-	c.retryCollectors(nil)
+	c.retryCollectors(nil, optional.NewNoneOption[workloadmeta.Component]())
```

Comment:
instead of providing the `workloadmeta.Component` to the `CollectorRegistry.retryCollectors` why we don't add it to the CollectorRegistry struct and change the function `newCollectorRegistry()` to provide it as parameter?

Reply:
I tried what you suggest during the development but `newCollectorRegistry` is called from an [init function](https://github.com/DataDog/datadog-agent/blob/4293b682aff2d870bddf0d0fc562db4f21bc80e7/pkg/util/containers/metrics/provider/registry.go#L151-L155)

Reply:
ok make sense,
we need to make this `CollectorRegistry` a component too 😓